src/main/kotlin/sliv/tool/scene/SceneFacade.kt

```diff
@@ -0,0 +1,78 @@
+package sliv.tool.scene
+
+import javafx.scene.image.Image
+import sliv.tool.project.model.*
+import sliv.tool.scene.controller.SceneController
+import sliv.tool.scene.model.*
+import java.io.FileInputStream
+
+class SceneFacade(private val controller: SceneController) {
+    private val visualizationLayers = HashMap<String, Layer>()
+
+    fun visualize(layers: List<ProjectLayer>, frames: List<ProjectFrame>) {
+        layers.forEach { projectLayer ->
+            val visualizationLayer = visualizationLayers[projectLayer.name]
+            if (visualizationLayer != null && visualizationLayer.kind == projectLayer.kind) {
+                return //don't rewrite existing settings
```

Comment:
you return from the whole function? I think forEach tag is skipped

Reply:
why you check the kind? lets think that names are unique) than you can use `?:` operator here

Reply:
Project was re-imported -> layer "ORB-points" somehow now is for planes -> CRASH

So I just rewrite it in dictionary